org.openjdk.jmc.console.ext.agent/src/main/java/org/openjdk/jmc/console/ext/agent/manager/wizards/PresetEditingWizardConfigPage.java

```diff
@@ -65,13 +84,183 @@ public void createControl(Composite parent) {
 		Composite container = new Composite(sc, SWT.NONE);
 		sc.setContent(container);
 
-		// TODO: create manager control here
-		container.setLayout(new FillLayout());
-		new Label(container, SWT.NONE).setText("TODO: create config page control here");
+		GridLayout layout = new GridLayout();
+		container.setLayout(layout);
+
+		{
+			Composite fileNameContainer = createFileNameContainer(container);
+			GridData gd = new GridData(SWT.FILL, SWT.FILL, true, false);
+			fileNameContainer.setLayoutData(gd);
+		}
+
+		{
+			Control separator = createSeparator(container);
+			GridData gd = new GridData(SWT.FILL, SWT.FILL, true, false);
+			separator.setLayoutData(gd);
+		}
+
+		{
+			Composite globalConfigContainer = createGlobalConfigContainer(container);
+			GridData gd = new GridData(SWT.FILL, SWT.FILL, true, false);
+			globalConfigContainer.setLayoutData(gd);
+		}
+
+		populateUi();
+		bindListeners();
 
 		sc.setExpandHorizontal(true);
 		sc.setExpandVertical(true);
 		sc.setMinSize(container.computeSize(SWT.DEFAULT, SWT.DEFAULT));
 		setControl(sc);
 	}
+
+	private Composite createFileNameContainer(Composite parent) {
+		Composite container = new Composite(parent, SWT.NONE);
+		int cols = 5;
+		GridLayout layout = new GridLayout(cols, false);
+		layout.horizontalSpacing = 8;
+		container.setLayout(layout);
+
+		createFileNameInput(container, cols);
+
+		return container;
+	}
+
+	private void createFileNameInput(Composite parent, int cols) {
+		Label label = createLabel(parent, LABEL_FILE_NAME);
+		label.setLayoutData(new GridData(SWT.BEGINNING, SWT.CENTER, false, false));
+
+		fileNameText = createText(parent);
+		GridData gd = new GridData(SWT.FILL, SWT.CENTER, true, true);
+		gd.horizontalSpan = cols - 2;
+		gd.minimumWidth = 0;
+		gd.widthHint = 400;
+		fileNameText.setLayoutData(gd);
+	}
+
+	private void setErrorMessageIfAny() {
+		if (fileNameError != null) {
+			setErrorMessage(fileNameError.getMessage());
+			return;
+		}
+
+		if (classPrefixError != null) {
+			setErrorMessage(classPrefixError.getMessage());
+			return;
+		}
+
+		setErrorMessage(null);
+	}
+
+	private Composite createGlobalConfigContainer(Composite parent) {
+		Composite container = new Composite(parent, SWT.NONE);
+		int cols = 5;
+		GridLayout layout = new GridLayout(cols, false);
+		layout.horizontalSpacing = 8;
+		container.setLayout(layout);
+
+		Label globalConfigLabel = new Label(container, SWT.NONE);
+		globalConfigLabel.setText(LABEL_GLOBAL_CONFIG);
```

Comment:
set globalConfigLabel message here as well

Reply:
Sorry, I'm a bit confused. This is a label, not a text input.

Reply:
oops I meant to add this comment to the classPrefixText at line [178](https://github.com/rh-jmc-team/jmc-agent-plugin/pull/62/files/e368ee9b044a4a58e69768c638948c74ea84476d#diff-9bdc9ceb1c34e29570f65d8953cf435cR178)